client/src/components/ResultsItem/index.js

```diff
@@ -1,17 +1,19 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import noop from 'lodash.noop';
+import { withRouter } from 'react-router-dom';
+import ReactRouterPropTypes from 'react-router-prop-types';
 import MovieSchema from '../../constants/MovieSchema';
 import './styles.scss';
 
-const ResultsItem = ({ movie, onSelectHandler }) => {
+const ResultsItem = ({ movie, onSelectHandler, history }) => {
     const {
         poster_path: path, title, genres, release_date: date
     } = movie;
 
     return (
         <div className="results-item">
-            <img className="results-item__poster" src={path} alt={title} role="presentation" onClick={() => onSelectHandler(movie)} />
+            <img className="results-item__poster" src={path} alt={title} role="presentation" onClick={() => { onSelectHandler(movie); history.push(`/film/${movie.id}`); }} />
```

Comment:
You have 2 better options here:
1. use `NavLink`
2. make ResultsItem a class and make this handler a class property